tests/test_inngest/test_client/test_send.py

```diff
@@ -130,6 +132,229 @@ async def test_partial_send_error(self) -> None:
             )
         assert len(ctx.exception.ids) == 1
 
+    async def test_client_send_retry(self) -> None:
+        """
+        Ensure that the client retries on error when sending to event API, and
+        that retries use the same idempotency key header.
+        """
+
+        test_options: typing.List[
+            typing.Tuple[
+                typing.Callable[
+                    [], typing.Tuple[http_proxy.OnRequest, CountGetter]
+                ],
+                bool,
+            ]
+        ] = [
+            (create_first_request_500_handler, True),
+            (create_first_request_timeout_handler, True),
+            (create_first_request_500_handler, False),
+            (create_first_request_timeout_handler, False),
+        ]
+
+        for proxy_request_handler_factory, is_sync in test_options:
+            with self.subTest(proxy_request_handler_factory):
+                on_request, get_proxy_request_counter = (
+                    proxy_request_handler_factory()
+                )
```

Comment:
I ended up doing this to parameterize the test instead of the `base.Case` example in https://github.com/inngest/inngest-py/blob/main/tests/test_inngest/test_function/cases/ai_infer.py#L138-L166, since that seemed a bit more heavyweight and necessary for testing with the different web frameworks. using `subTest` allows us to still run a single test a la `pytest -n 4 -v ./test_client/test_send.py -k test_client_send_retry`